7_drivers_munge/src/GCM_driver_utils.R

```diff
@@ -191,18 +202,25 @@ adjust_lake_cell_tile_xwalk <- function(spatial_xwalk, lake_centroids, query_cel
   cell_centroids_with_data <- query_cell_centroids %>%
     filter(cell_no %in% cells_with_data)
 
-  # match each lake to a cell that returned data based on
-  # which cell centroid is closest to each lake centroid
-  lake_cells_with_data_tiles_xwalk <- lake_centroids %>%
-    st_join(cell_centroids_with_data, join=st_nearest_feature, left=TRUE) %>%
-    st_drop_geometry() %>%
-    select(site_id, state, cell_no, tile_no)
-
-  # join this adjusted xwalk to the spatial xwalk
-  adjusted_xwalk <- spatial_xwalk %>%
-    left_join(lake_cells_with_data_tiles_xwalk,
-              by=c('site_id', 'state'),
-              suffix=c('_spatial','_data'))
+  # match each lake to a cell that returned data, preferably to a cell 1) within the same row as, and
+  # 2) within `x_buffer` x distance (number of columns) of, the cell within which the lake falls spatially
+  # If no non-nan cells meet that criteria, match instead to the closest non-nan cell, regardless of row
+  adjusted_xwalk <- lake_centroids %>%
+    left_join(spatial_xwalk, by=c('site_id', 'state')) %>% # add cell_no, tile_no for each lake
+    left_join(query_cell_centroids %>% st_drop_geometry(), by=c('cell_no','tile_no')) %>% # add x, y coords of cell_no for each lake
+    rename(x_spatial=x, y_spatial=y, cell_no_spatial=cell_no, tile_no_spatial=tile_no) %>%
+    group_by(y_spatial, x_spatial) %>%
```

Comment:
So you are grouping by cell here, correct? So lakes within the same NA cell would end up getting the same new cell regardless of where in that cell the lakes were, correct? This seems like a slight deviation of what you had implemented previously.

Reply:
Yay! Glad that was confirmed